pkg/apis/openebs.io/scp/v1alpha1/storage-cohort.go

```diff
@@ -17,9 +17,16 @@
 package v1alpha1
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
+	// StoragePoolProtectionFinalizer makes sure that no pool is present
+	// at the time of storage cohort CR deletion
+	StoragePoolProtectionFinalizer = "openebs.io/pool-protection"
```

Comment:
IMO, we should have a single deletion finalizer, the plug-in handles the implementation.
cc @shovanmaity